src/provider/systemdmanager/manager.rs

```diff
@@ -0,0 +1,262 @@
+        &self,
+        name: &str,
+        unit: &SystemDUnit,
+        unit_type: UnitTypes,
+    ) -> Result<(), anyhow::Error> {
+        // Appends .service to name if necessary
+        let unit_name = Manager::get_unit_file_name(name, unit_type)?;
+
+        let unit_path = PathBuf::from(&unit_name);
+
+        // Check if an absolute path was provided
+        // if this is an absolute path the file will be written there, otherwise
+        // the file will be created in the unit directory
+        let target_file = if unit_path.is_absolute() {
+            unit_path
+        } else {
+            self.units_directory.join(unit_path)
+        };
+
+        //let target_file = self.units_directory.join(&unit_name);
+        debug!("Target file for service {} : {:?}", name, target_file);
+
+        let mut unit_file = match File::create(&target_file) {
+            Ok(file) => file,
+            Err(e) => {
+                error!("Error ocurred when creating unit file [{}]: [{}]", name, e);
+                return Err(anyhow::Error::from(e));
+            }
+        };
+
+        unit_file.write_all(unit.get_unit_file_content().as_bytes())?;
+        unit_file.flush()?;
+
+        self.systemd_connection
+            .enable_unit_file(&target_file.into_os_string().to_string_lossy())?;
+        self.reload()?;
+
+        Ok(())
+    }
+
+    // Stop and disable the service, then delete the unit file from disk
+    pub fn unload(&self, name: &str, unit_type: UnitTypes) -> Result<(), anyhow::Error> {
+        // Appends .service to name if necessary
+        let unit_name = Manager::get_unit_file_name(name, unit_type)?;
+
+        self.systemd_connection
+            .disable_unit_file(&unit_name)
+            .map_err(anyhow::Error::from)
+    }
+
+    pub fn start(&self, name: &str) -> Result<(), anyhow::Error> {
+        let unit_name = Manager::get_unit_file_name(name, UnitTypes::Service)?;
+        match self.systemd_connection.start_unit(&unit_name) {
+            Ok(result) => info!("Successfully started service [{}]: [{}]", unit_name, result),
+            Err(e) => {
+                error!("Error: [{}]", e);
+                return Err(anyhow!("Error starting service [{}]: {}", name, e));
+            }
+        };
+        Ok(())
+    }
+
+    pub fn stop(&self, name: &str) -> Result<(), anyhow::Error> {
+        let unit_name = Manager::get_unit_file_name(name, UnitTypes::Service)?;
+        match self.systemd_connection.stop_unit(&unit_name) {
+            Ok(result) => info!("Successfully stopped service [{}]: [{}]", unit_name, result),
+            Err(e) => {
+                error!("Error: [{}]", e);
+                return Err(anyhow!("Error stopping service [{}]: {}", name, e));
+            }
+        };
+        Ok(())
+    }
+
+    /// TODO: Make this actually do something, currently it only returns true
+    #[allow(unused_variables)]
+    pub fn is_active(&self, name: &str) -> Result<bool, anyhow::Error> {
```

Comment:
I am actually not sure yet how this entire thing will work, this was just a try at a quick-win instead of subscribing to the events of our service.
I have removed the entire code for now, as we are not targeting this functionality for milestone 1 anyway.